### DISCUSSION TITLE
Discrepancy in Nuclei templates updation

### DISCUSSION THREAD

**Discussion by tarunkant** [projectdiscovery/nuclei-templates#4639] [Q&A] [answered]
Hello, as per the nuclei get-started guide(https://nuclei.projectdiscovery.io/nuclei/get-started/#nuclei-templates), it states that:
> You may still use the update-templates flag to update the nuclei templates at any time; automatic updates happen every **24** hours.

But I wanted to ask if nuclei templates get updated automatically in 24 hrs? Because `nuclei -update-templates` works on the release version of nuclei-templates but it was noticed that the release doesn't happen every day, Ref. https://github.com/projectdiscovery/nuclei-templates/releases
Notice the release time difference between v9.0.7 and v9.0.8 which is around 8 days. It means for 8 days running `nuclei -update-templates` is not updating the database as no new release happened. However, it was stated that the nuclei-templates get updated automatically in 24 hrs, which is creating confusion.

It would be great if you could clarify this, thank you!

**Comment by ehsandeep** [comment]
Hi @tarunkant,

I've removed the discussion from the nuclei project as it was duplicate and [updated](https://github.com/projectdiscovery/nuclei-docs/commit/c8b8955f4c70eeed34a1f1cecf2adb1f2e64d1f0) the nuclei-docs as it was outdated which caused the confusion.

As per the update which should be reflected at https://nuclei.projectdiscovery.io/nuclei/get-started/#nuclei-templates

> Nuclei checks for new template releases upon each execution and update to the latest version automatically if updates are available. Optionally automatic update/check can be disabled using `-duc, -disable-update-check` CLI flag or flag config file.

**Comment by tarunkant** [comment, marked as answer]
@ehsandeep, to get the new templates added on daily basis is only through git pull, then? As `-update-templates` updates templates using release version and release may take time as we have already seen in the https://github.com/projectdiscovery/nuclei-templates/releases.

**Reply by ehsandeep** [reply]
@tarunkant right, nuclei built-in option download / update the released version of templates, if you wish to use templates added in master, you have to manually clone the template project and provide cloned template directory to nuclei, instead the using the default one.

**Reply by tarunkant** [reply]
Understood, It clarifies my doubt. Thank you so much!
